### ISSUE TITLE
Feature: experiment with pyqt multimedia

### ISSUE THREAD

**Issue by achille-martin** [achille-martin/pyqt-crom#23]
The aim of this issue is to experiment with PyQt multimedia features like: Video, Screenshot, Image,...

**Comment by achille-martin** [comment]
Issue with screenshot (QScreen.grabWindow() returning an empty QPixmap) on Android but not on Linux:

```
[2023-11-16 02:27:48,849] [DEBUG] - MainWindow::grab_screen <(
[2023-11-16 02:27:48,849] [INFO] - MainWindow::grab_screen - Grabbing a screenshot
[2023-11-16 02:27:48,850] [DEBUG] - MainWindow::grab_screen - Device 1 object confirmation: <PyQt5.QtGui.QScreen object at 0x7efd60656f78>
[2023-11-16 02:27:48,850] [DEBUG] - MainWindow::grab_screen - Screenshot 1 object confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd604e8908>
[2023-11-16 02:27:48,850] [DEBUG] - MainWindow::grab_screen - Screenshot 1 dimensions: 0 x 0
[2023-11-16 02:27:48,851] [DEBUG] - MainWindow::grab_screen - Device 2 object confirmation: <PyQt5.QtGui.QWindow object at 0x7efd6065aaf8>
[2023-11-16 02:27:48,851] [DEBUG] - MainWindow::grab_screen - Screenshot 2 object confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd604e8a58>
[2023-11-16 02:27:48,851] [DEBUG] - MainWindow::grab_screen - Screenshot 2 dimensions: 0 x 0
[2023-11-16 02:27:48,852] [DEBUG] - MainWindow::grab_screen - Device 3 object confirmation: None
[2023-11-16 02:27:48,852] [DEBUG] - MainWindow::grab_screen - Screenshot 3 object confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd604e8ba8>
[2023-11-16 02:27:48,852] [DEBUG] - MainWindow::grab_screen - Screenshot 3 dimensions: 0 x 0
[2023-11-16 02:27:48,853] [DEBUG] - MainWindow::grab_screen - Device 4 object confirmation: <PyQt5.QtGui.QScreen object at 0x7efd60656f78>
[2023-11-16 02:27:48,853] [DEBUG] - MainWindow::grab_screen - Screenshot 4 object confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd604e8c88>
[2023-11-16 02:27:48,853] [DEBUG] - MainWindow::grab_screen - Screenshot 4 dimensions: 0 x 0
[2023-11-16 02:27:48,854] [DEBUG] - MainWindow::grab_screen - Device 5 object confirmation: <PyQt5.QtGui.QWindow object at 0x7efd6065e678>
[2023-11-16 02:27:48,854] [DEBUG] - MainWindow::grab_screen - Screenshot 5 object confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd604e8f28>
[2023-11-16 02:27:48,854] [DEBUG] - MainWindow::grab_screen - Screenshot 5 dimensions: 0 x 0
[2023-11-16 02:27:48,855] [DEBUG] - MainWindow::grab_screen - Device 6 object confirmation: <PyQt5.QtGui.QScreen object at 0x7efd60656f78>
[2023-11-16 02:27:48,855] [DEBUG] - MainWindow::grab_screen - Screenshot 6 object confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd5d72bba8>
[2023-11-16 02:27:48,856] [DEBUG] - MainWindow::grab_screen - Screenshot 6 dimensions: 0 x 0
[2023-11-16 02:27:48,856] [DEBUG] - MainWindow::grab_screen - Device 7 all screens: [<PyQt5.QtGui.QScreen object at 0x7efd60656f78>]
[2023-11-16 02:27:48,856] [DEBUG] - MainWindow::grab_screen - Device 7 all windows: [<PyQt5.QtGui.QWindow object at 0x7efd6065e4c8>, <PyQt5.QtGui.QWindow object at 0x7efd605e24c8>, <PyQt5.QtGui.QWindow object at 0x7efd605e2e58>, <PyQt5.QtGui.QWindow object at 0x7efd6065e678>, <PyQt5.QtGui.QWindow object at 0x7efd6065aaf8>]
[2023-11-16 02:27:48,857] [DEBUG] - MainWindow::grab_screen - Device 7 screen confirmation: <PyQt5.QtGui.QScreen object at 0x7efd60656f78>
[2023-11-16 02:27:48,857] [DEBUG] - MainWindow::grab_screen - Device 7 screen size confirmation: PyQt5.QtCore.QSize(1080, 2138)
[2023-11-16 02:27:48,858] [DEBUG] - MainWindow::grab_screen - Window 0 object: <PyQt5.QtGui.QWindow object at 0x7efd6065e4c8>
[2023-11-16 02:27:48,858] [DEBUG] - MainWindow::grab_screen - Window 0 dimensions: 640 x 480
[2023-11-16 02:27:48,858] [DEBUG] - MainWindow::grab_screen - Window 0 QPixmap confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd5d72b278>
[2023-11-16 02:27:48,858] [DEBUG] - MainWindow::grab_screen - Window 0 dimensions: PyQt5.QtCore.QSize()
[2023-11-16 02:27:48,859] [DEBUG] - MainWindow::grab_screen - Image 7 dimensions: 0 x 0
[2023-11-16 02:27:48,859] [DEBUG] - MainWindow::grab_screen - Window 1 object: <PyQt5.QtGui.QWindow object at 0x7efd605e24c8>
[2023-11-16 02:27:48,859] [DEBUG] - MainWindow::grab_screen - Window 1 dimensions: 1080 x 2138
[2023-11-16 02:27:48,859] [DEBUG] - MainWindow::grab_screen - Window 1 QPixmap confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd5d72b0b8>
[2023-11-16 02:27:48,860] [DEBUG] - MainWindow::grab_screen - Window 1 dimensions: PyQt5.QtCore.QSize()
[2023-11-16 02:27:48,860] [DEBUG] - MainWindow::grab_screen - Image 7 dimensions: 0 x 0
[2023-11-16 02:27:48,860] [DEBUG] - MainWindow::grab_screen - Window 2 object: <PyQt5.QtGui.QWindow object at 0x7efd605e2e58>
[2023-11-16 02:27:48,860] [DEBUG] - MainWindow::grab_screen - Window 2 dimensions: 1080 x 57
[2023-11-16 02:27:48,861] [DEBUG] - MainWindow::grab_screen - Window 2 QPixmap confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd5d72b128>
[2023-11-16 02:27:48,861] [DEBUG] - MainWindow::grab_screen - Window 2 dimensions: PyQt5.QtCore.QSize()
[2023-11-16 02:27:48,861] [DEBUG] - MainWindow::grab_screen - Image 7 dimensions: 0 x 0
[2023-11-16 02:27:48,862] [DEBUG] - MainWindow::grab_screen - Window 3 object: <PyQt5.QtGui.QWindow object at 0x7efd6065e678>
[2023-11-16 02:27:48,862] [DEBUG] - MainWindow::grab_screen - Window 3 dimensions: 1080 x 2081
[2023-11-16 02:27:48,862] [DEBUG] - MainWindow::grab_screen - Window 3 QPixmap confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd5d72b278>
[2023-11-16 02:27:48,862] [DEBUG] - MainWindow::grab_screen - Window 3 dimensions: PyQt5.QtCore.QSize()
[2023-11-16 02:27:48,863] [DEBUG] - MainWindow::grab_screen - Image 7 dimensions: 0 x 0
[2023-11-16 02:27:48,863] [DEBUG] - MainWindow::grab_screen - Window 4 object: <PyQt5.QtGui.QWindow object at 0x7efd6065aaf8>
[2023-11-16 02:27:48,863] [DEBUG] - MainWindow::grab_screen - Window 4 dimensions: 1080 x 2138
[2023-11-16 02:27:48,863] [DEBUG] - MainWindow::grab_screen - Window 4 QPixmap confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd5d72b0b8>
[2023-11-16 02:27:48,863] [DEBUG] - MainWindow::grab_screen - Window 4 dimensions: PyQt5.QtCore.QSize()
[2023-11-16 02:27:48,864] [DEBUG] - MainWindow::grab_screen - Image 7 dimensions: 0 x 0
[2023-11-16 02:27:48,864] [DEBUG] - MainWindow::grab_screen - Device 8 object confirmation: <PyQt5.QtGui.QScreen object at 0x7efd60656f78>
[2023-11-16 02:27:48,864] [DEBUG] - MainWindow::grab_screen - Screenshot 8 object confirmation: <PyQt5.QtGui.QPixmap object at 0x7efd5d72b128>
[2023-11-16 02:27:48,865] [DEBUG] - MainWindow::grab_screen - Screenshot 8 dimensions: 0 x 0
[2023-11-16 02:27:48,865] [DEBUG] - MainWindow::grab_screen - Total screenshots taken: 29
[2023-11-16 02:27:48,865] [DEBUG] - MainWindow::grab_screen )>
```

**Comment by achille-martin** [comment]
The issues seems to come from Qt itself according to [this bug report](https://bugreports.qt.io/browse/QTBUG-63275?jql=text%20~%20%22Grabwindow(0)%20android%22).

A fix has been implemented and tested for versions of Qt >= 5.15.4

Note that Qt version 5.15 LTS will be supported until mid 2025. This should promote the adoption of 5.15 instead of 5.12 for the next 2 years.

**ACTION**: move to Qt version 5.15.11 (latest update Oct 2023 according to the [list of official Qt releases](https://download.qt.io/official_releases/qt/5.15/)).

**Comment by achille-martin** [comment]
Challenge: PyQt5 v5.15.6 compatible with Python3.6 and released just after Qt5.15.2 (and before Qt5.15.3).
Therefore, it might only be possible to install Qt5.15.2 via the online installer and not be possible to install Qt > Qt5.15.4...
